### PR TITLE
Update CanaryServiceInstanceFilter.java

### PR DESCRIPTION
**Pull request by zmz**
support allowCrossApp canary release

### CODE REVIEW

File: spring-cloud-huawei-router-client/src/main/java/com/huaweicloud/router/client/loabalancer/CanaryServiceInstanceFilter.java [hunk 1]
```diff
@@ -57,6 +57,12 @@ public CanaryServiceInstanceFilter(AbstractRouterDistributor<ServiceInstance, Mi
   public List<ServiceInstance> filter(ServiceInstanceListSupplier supplier, List<ServiceInstance> instances,
       Request<?> request) {
     String targetServiceName = supplier.getServiceId();
+    String[] tmpList = targetServiceName.split("\\"+DiscoveryConstants.APP_SERVICE_SEPRATOR);
+
+    if (tmpList.length >= 2) {
+      targetServiceName = tmpList[1];
+    }
```
Comment (liubao68) [inline spring-cloud-huawei-router-client/src/main/java/com/huaweicloud/router/client/loabalancer/CanaryServiceInstanceFilter.java:64]: Can you create an issue and give some description about the problem?

Comment (zmz) [reply spring-cloud-huawei-router-client/src/main/java/com/huaweicloud/router/client/loabalancer/CanaryServiceInstanceFilter.java:64]: Called across apps, the canary rule does not take effect
例如，across app called, targetServiceName = app.servicename（E.g  app.canary-provider）
RouterFilter的routerRuleCache.doInit will go wrong

canary rule is：
servicecomb:
  routeRule:
    canary-provider: |
      - precedence: 2 #优先级
        match:        #匹配策略
          headers:          #header匹配
            tenant:
              exact: '10000'
        route: #路由规则
          - weight: 100
            tags:
              version: 0.0.2

**Comment by liubao68** [comment]
Seams duplicate with https://github.com/huaweicloud/spring-cloud-huawei/pull/676

**Comment by liubao68** [comment]
see https://github.com/huaweicloud/spring-cloud-huawei/issues/669

**Pull request closed without merging**